plasmapy/formulary/shocks.py

```diff
@@ -0,0 +1,122 @@
+""" This module gathers functions relating to shocks and the properties thereof.
+"""
+__all__ = ["entropy_jump_polytropic"]
+
+
+from astropy import units as u
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log, nan, isclose
+from warnings import warn
+
+from plasmapy.utils.exceptions import PhysicsWarning
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+
+
+@validate_quantities(
+    rho_1={"can_be_negative": False, "pass_equivalent_units": True},
+    rho_2={"can_be_negative": False, "pass_equivalent_units": True},
+    p_1={"can_be_negative": True, "pass_equivalent_units": True},
+    p_2={"can_be_negative": True, "pass_equivalent_units": True},
+    c_v={"can_be_negative": False},
+)
+def entropy_jump_polytropic(
+    c_v: u.J / u.K,
+    p_1: u.Pa,
+    p_2: u.Pa,
+    rho_1: u.kg / u.m ** 3,
```

Comment:
The choice of mass density vs. number density is definitely field-specific: I think using rho is more common in ICF while in space physics/lab astro I see number density more. I'll just point out that if you do use a `Particle` input for the species, this _could_ be changed to number density and then converted to mass density within the function.

Reply:
I think you can just get away with annotating with `[u.kg / u.m ** 3, u.m ** -3]`.  This should allow either number density or mass density to pass through. Then below when you to calculate the ratio you can do something like...

```python
try:
    pressure_ratio = (p_2 / p_1).to(u.dimensionless_unscaled)
    density_ratio = (rho_1 / rho_2).to(u.dimensionless_unscaled)
except u.UnitConversionError as err:
    raise u.UnitConversionError(
        f"Upstream and downstream parameters must have the same units, {err}"
    )
```

Reply:
^Yes, this is the right answer.

Reply:
I'd like to note...Drake's derivation of this uses mass density for rho.  Since I can't think of a scenario where you'd have two different species for upstream/downstream, the mass ratio is equal to the number ratio.